### PR TITLE
feat(gatsby-recipes): add Chakra UI recipe

### PR DESCRIPTION
**Pull request by shoxton**
## Description
This PR adds [Chakra UI](https://github.com/chakra-ui/chakra-ui) to the recipes list

### Documentation
You can read more about this recipe documentation [here](https://github.com/shoxton/gatsby-recipe-chakra-ui)

### CODE REVIEW

**Comment by KyleAMathews** [comment]
Hey! Thanks for adding this. Tested it out and found a couple of issues. One of which was a bug in recipes which I fixed in https://github.com/gatsbyjs/gatsby/pull/27790

The second is when I did get the recipe to run with the above fix, things didn't work out-of-the-box — looking at the Chakra docs, we also need to install 3 emotion dependencies — https://github.com/chakra-ui/chakra-ui#installing-chakra-ui

**Comment by shoxton** [comment]
> Hey! Thanks for adding this. Tested it out and found a couple of issues. One of which was a bug in recipes which I fixed in #27790
> 
> The second is when I did get the recipe to run with the above fix, things didn't work out-of-the-box — looking at the Chakra docs, we also need to install 3 emotion dependencies — https://github.com/chakra-ui/chakra-ui#installing-chakra-ui

Thanks for checking my code. I think i fixed this on https://github.com/gatsbyjs/gatsby/pull/27721/commits/5c995243a8037933ebe723589a60371ef4f41927 so i changed the title to be more clear.

I replaced the shadow logic which was using the `<GatsbyShadowFile>` api by the `<File>` api to cover the bug you mentioned, just like [Theme UI recipe](https://raw.githubusercontent.com/gatsbyjs/gatsby/master/packages/gatsby-recipes/recipes/theme-ui.mdx) does. I also added the missing packages.

If theres anything that im missing out, please let me know.

**Comment by KyleAMathews** [comment]
Oh can you revert to use ShadowFile? The bug will be fixed shortly.

File: packages/gatsby-recipes/recipes/chakra-ui.mdx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+## Installs `gatsby-plugin-chakra-ui` in `gatsby-config.js`
+
+<GatsbyPlugin name="gatsby-plugin-chakra-ui" />
+
+---
+
+## Creates gatsby-plugin-chakra-ui `theme.js` file so you can customize your own global styles
+
+<File
```
Comment (KyleAMathews) [inline packages/gatsby-recipes/recipes/chakra-ui.mdx:27]: The fix is in

```suggestion
<GatsbyShadowFile theme="gatsby-plugin-chakra-ui" path="src/theme.js" />
```

**Review by KyleAMathews — APPROVED** [review]
Thanks!

**Comment by shoxton** [comment]
My pleasure! Is it possible to add this recipe to the official menu list as stated on https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-recipes/recipes#gatsby-recipes-mdx-source ?
Thanks!

**Comment by KyleAMathews** [comment]
Oh of course. I overlooked that. Please PR it to
https://github.com/gatsbyjs/gatsby/blob/d0c94e98f013e53d946e0e75a3e4c1df20ae4c04/packages/gatsby-recipes/src/recipes-list.js

--
Kyle Mathews

Blog: *http://bricolage.io <http://bricolage.io>*
Twitter: http://twitter.com/kylemathews

**Comment by shoxton** [comment]
Done! Just created a PR #28013 to add this recipe to the list

**Pull request merged**